EC-84: Revise Configuration, Customers, eContent, Login and Logging (#5)
ezsystems/commerce-documentation: docs/guide/econtent/econtent_features/catalog_segmentation.md

## Custom segmentation logic

If you need custom logic, override `ConfigEcontentCatalogSegmentationService` to return the catalog codes
which are used in your context (e.g. depending on the user). 

Catalog segmentation uses a service to define which users are allowed to see the content specified in segmentated parts of the catalog.

The default configuration uses `ConfigEcontentCatalogSegmentationService`
that returns a collection of catalog codes specified in the configuration.
The service implements `EcontentCatalogSegmentationServiceInterface`.

You can implement this interface when you need a custom algorithm
for showing specific products to specific users.

The following example service replaces the default implementation:

``` php
<?php

namespace Silversolutions\Bundle\EshopBundle\Service;

use eZ\Publish\Core\MVC\ConfigResolverInterface;
use Silversolutions\Bundle\EshopBundle\Api\EcontentCatalogSegmentationServiceInterface;

/**
 * Simple user group based implementation.
 */
class MyCustomUserEcontentCatalogSegmentationService implements EcontentCatalogSegmentationServiceInterface
{
    /**
     * @var string
     */
    private $userGroup;

    /**

>      * Set via external instance depending to request session

Reviewer: What does the sentence mean? Is it ", depending on the requesting session"?

     *
     * @param string $groupCode
     */
    public function setUserGroup($groupCode)
    {
        $this->userGroup = $groupCode;
    }

    /**

>      * The logic to fetch the desired catalog codes should be placed here:

Reviewer: ```suggestion
     * Place the logic that fetches the desired catalog codes here:
```
